### PR TITLE
unpin peer dependencies

### PR DESCRIPTION
**Pull request by julien51**
# Description

Not pinning the peedependencies and changing the renovate rule.

# Checklist:

- [X] 1 PR, 1 purpose: my Pull Request applies to a single purpose
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the docs to reflect my changes if applicable
- [ ] I have added tests (and stories for frontend components) that prove my fix is effective or that my feature works
- [ ] I have performed a self-review of my own code
- [ ] If my code involves visual changes, I am adding applicable screenshots to this thread

### CODE REVIEW

File: packages/ui/package.json [hunk 1]
```diff
@@ -44,9 +44,8 @@
     "tailwindcss": "3.2.4"
   },
   "peerDependencies": {
-    "@headlessui/react": "1.7.4",
-    "react": "17.0.2",
-    "react-dom": "17.0.2"
+    "^react": "17.0.2",
+    "^react-dom": "17.0.2"
```
Comment (searchableguy) [inline packages/ui/package.json:48]: Removes headless ui and adds ^ in the package, not the version?

File: packages/unlock-js/package.json [hunk 2]
```diff
@@ -28,7 +28,7 @@
   "author": "Unlock Inc",
   "license": "MIT",
   "peerDependencies": {
-    "ethers": "5.7.2"
+    "ethers": "^5.7.2"
```
Comment (searchableguy) [inline packages/unlock-js/package.json:31]: Also we could do ^5.x.x so so we support any 5 version of ethers. 

Depends on what features we use internally of ethers.

File: packages/ui/package.json (hunk 1 above)
Comment (julien51) [reply packages/ui/package.json:48]: good catch!

File: packages/unlock-js/package.json (hunk 2 above)
Comment (julien51) [reply packages/unlock-js/package.json:31]: fixed!

File: packages/hardhat-plugin/package.json [hunk 3]
```diff
@@ -51,8 +51,8 @@
     "typescript": "4.9.3"
   },
   "peerDependencies": {
-    "ethers": "5.7.2",
-    "hardhat": "2.12.3"
+    "ethers": "^5.7",
```
[the remark this answers is not in the document]
Comment (julien51) [reply packages/hardhat-plugin/package.json:54]: Isn't that equivalent?

File: packages/hardhat-plugin/package.json [hunk 4]
```diff
@@ -51,8 +51,8 @@
...
+    "hardhat": "^2.12"
```
Comment (julien51) [inline packages/hardhat-plugin/package.json:55]:
```suggestion
    "hardhat": "^2.12.x"
```

File: packages/jest-config/package.json [hunk 5]
```diff
@@ -11,6 +11,6 @@
     "ts-jest": "27.1.5"
   },
   "peerDependencies": {
-    "jest": "27.5.1"
+    "jest": "^27.5"
```
Comment (julien51) [inline packages/jest-config/package.json:14]:
```suggestion
    "jest": "^27.5.x"
```

File: packages/ui/package.json [hunk 6]
```diff
@@ -44,9 +44,8 @@
     "tailwindcss": "3.2.4"
   },
   "peerDependencies": {
-    "@headlessui/react": "1.7.4",
-    "react": "17.0.2",
-    "react-dom": "17.0.2"
+    "react": "^17",
```
Comment (julien51) [inline packages/ui/package.json:47]:
```suggestion
    "react": "^17.x.x",
```

File: packages/ui/package.json [hunk 7]
```diff
@@ -44,9 +44,8 @@
...
+    "react-dom": "^17"
```
Comment (julien51) [inline packages/ui/package.json:48]:
```suggestion
    "react-dom": "^17.x.x"
```

File: packages/unlock-assets/package.json [hunk 8]
```diff
@@ -32,7 +32,7 @@
     "build-storybook": "build-storybook"
   },
   "peerDependencies": {
-    "react": "17.0.2"
+    "react": "^17"
```
Comment (julien51) [inline packages/unlock-assets/package.json:35]:
```suggestion
    "react": "^17.x.x"
```

File: packages/unlock-express/package.json [hunk 9]
```diff
@@ -14,7 +14,7 @@
     "passport-custom": "1.1.1"
   },
   "peerDependencies": {
-    "passport": "0.6.0"
+    "passport": "^0.6"
```
Comment (julien51) [inline packages/unlock-express/package.json:17]:
```suggestion
    "passport": "^0.6.x"
```

File: packages/unlock-js/package.json [hunk 10]
```diff
@@ -28,7 +28,7 @@
   "author": "Unlock Inc",
   "license": "MIT",
   "peerDependencies": {
-    "ethers": "5.7.2"
+    "ethers": "^5.7"
```
Comment (julien51) [inline packages/unlock-js/package.json:31]:
```suggestion
    "ethers": "^5.7.x"
```

**Review by searchableguy — APPROVED** [review]

**Pull request merged**